src/citrine/resources/table.py

```diff
@@ -144,6 +152,104 @@ def build_versions(collection: Iterable[dict]) -> Iterable[Table]:
+        ----------
+        config:
+            The persisted table config from which to build a table (or its ID).
+        version
+            The version of the table config, only necessary when config is a uid.
+        timeout
+            Amount of time to wait on build job (in seconds) before giving up. Defaults
+            to 15 minutes. Note that this number has no effect on the build job itself,
+            which can also time out server-side.
+
+        Returns
+        -------
+        Table
+            A new table built from the supplied config.
+
+        """
+        submission: JobSubmissionResponse = self._initiate_build_from_config(config, version)
+        logger.info('Build job submitted with job ID {}.'.format(submission.job_id))
+        path = 'projects/{}/execution/job-status'.format(self.project_id)
+        params = {'job_id': submission.job_id}
+        start_time = time()
+        while True:
+            response = self.session.get_resource(path=path, params=params)
+            status: JobStatusResponse = JobStatusResponse.build(response)
+            if status.status in ['Success', 'Failure']:
+                break
+            elif time() - start_time < timeout:
+                logger.info('Build job still in progress, polling status again in 2 seconds.')
+                sleep(2)
+            else:
+                logger.error('Build job exceeded user timeout of {} seconds.'.format(timeout))
+                logger.debug('Last status: {}'.format(status.dump()))
+                raise TimeoutError('Build job {} timed out.'.format(submission.job_id))
+        if status.status == 'Failure':
+            logger.debug('Job terminated with Failure status: {}'.format(status.dump()))
+            for task in status.tasks:
+                if task.status == 'Failure':
+                    logger.error('Task {} failed with reason "{}"'.format(
+                        task.id, task.failure_reason))
+            raise RuntimeError('Job {} terminated with Failure status.'.format(submission.job_id))
+        else:
+            table_id = status.output['display_table_id']
+            table_version = status.output['display_table_version']
+            warning_blob = status.output.get('table_warnings')
+            warnings = json.loads(warning_blob) if warning_blob is not None else []
+            if warnings:
+                warn_str = 'Table build completed with warnings:'
+                for warning in warnings:
+                    limited_results = warning.get('limited_results', [])
+                    for full_warning in limited_results:
+                        warn_str += '\n\t' + full_warning
+                    total_count = warning.get('total_count', 0)
+                    if total_count > len(limited_results):
+                        warn_str += '\n\t' + 'and {} more similar.'\
+                            .format(total_count - len(limited_results))
+                logger.warning(warn_str)
```

Comment:
Seems like repeated appending is better done as a join (untested):

```suggestion
                clauses = ['Table build completed with warnings:']
                for warning in warnings:
                    limited_results = warning.get('limited_results', [])
                    clauses.extend(limited_results)
                    excess = warning.get('total_count', 0) - len(limited_results)
                    if excess > 0:
                        clauses.append('and {} more similar.'.format(excess))
                logger.warning('\n\t'.join(clauses))
```